casbin/util/built_in_functions.cpp

```diff
@@ -140,6 +205,58 @@ bool KeyMatch3(const std::string& key1, const std::string& key2) {
     return res;
 }
 
+// KeyGet3 returns value matched pattern
+// For example, "project/proj_project1_admin/" matches "project/proj_{project}_admin/"
+// if the pathVar == "project", then "project1" will be returned
+std::string KeyGet3(const std::string& key1, const std::string& key2, const std::string& path_var) {
+    std::vector<std::string> key1_arr = Split(key1, "/");
+    std::vector<std::string> key2_arr = Split(key2, "/");
+
+    bool res = true;
+    std::string result = "";
+    for (int i = 0; i < key2_arr.size(); i++) {
+        if (i >= key1_arr.size()) {
+            res = false;
+            break;
+        }
+        if (key1_arr[i] != key2_arr[i]) {
+            size_t index1 = key2_arr[i].find("*");
+            size_t index2 = key2_arr[i].find("{");
+            size_t index3 = key2_arr[i].find("}");
+            if (index1 != std::string::npos) {
+                if (index1 == 0) {
+                    res = true;
+                    break;
+                } else if (key1_arr[i].compare(key2_arr[i].substr(0, index1))) {
+                    res = false;
+                    break;
+                } else
+                    continue;
+            }
+            if (index2 == 0 && index3 > 0 && index3 != std::string::npos) {
```

Comment:
How about `/admin_{res}` matching`/admin_data1` ?

Reply:
> How about `/admin_{res}` matching`/admin_data1` ?

For this example, KeyMatch3 will return false.